src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/InterventionController.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.InterventionDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.InterventionService
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.utils.StringToUUIDConverter
+
+@RestController
+class InterventionController(
+  private val interventionService: InterventionService,
+) {
+
+  @GetMapping("/intervention/{id}")
+  fun getInterventionByID(@PathVariable id: String): InterventionDTO {
+    val uuid = StringToUUIDConverter.parseID(id)
```

Comment:
Would it be possible to hook that conversion into the request mapper so we only have to do `getInterventionByID(@PathVariable id: UUID)`? It looks like that'd save a _lot_ of repetition in the controllers.

Reply:
Yes - updated

Reply:
this pattern is reused a lot. would be good to change all the other occurances (outside this PR).

Reply:
definitely